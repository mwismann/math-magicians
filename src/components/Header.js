import { NavLink } from 'react-router-dom';
import '../styles/Header.css';

const Header = () => (
  <header>
    <div className="logo-container">
      <h1>Math Magicians</h1>
    </div>
    <nav>
      <ul>
        <li><NavLink to="/">Home</NavLink></li>
        {' | '}
        <li><NavLink to="/calculator">Calculator</NavLink></li>
        {' | '}
        <li><NavLink to="/quote">Quote</NavLink></li>
      </ul>
    </nav>
  </header>
);

export default Header;
