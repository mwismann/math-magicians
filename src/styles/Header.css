header {
  margin: 0;
  padding: 0.75rem 2rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: rgb(39, 35, 43);
  border-bottom: 1px solid rgba(255, 255, 255, 0.18);
  box-shadow: 0 0.5rem 2rem 0 rgba(31, 38, 135, 0.37);
  backdrop-filter: blur(0.25rem);
  -webkit-backdrop-filter: blur(0.25rem);
}

.logo-container {
  padding: 0.5rem 2rem;
  border: 1px solid rgba(255, 255, 255, 0.18);
  border-radius: 1rem;
  box-shadow: 0 0.5rem 2rem 0 rgba(31, 38, 135, 0.37);
  backdrop-filter: blur(0.25rem);
  -webkit-backdrop-filter: blur(0.25rem);
}

h1 {
  margin: 0;
  color: rgb(200, 200, 200);
  text-shadow: 0 0.25rem rgba(31, 38, 135, 0.37);
}

nav {
  padding: 0.25rem 2rem;
  background-color: rgba(20, 21, 21, 0.25);
  border: 1px solid rgba(255, 255, 255, 0.18);
  border-radius: 1rem;
  box-shadow: 0 0.5rem 2rem 0 rgba(31, 38, 135, 0.37);
  backdrop-filter: blur(0.25rem);
  -webkit-backdrop-filter: blur(0.25rem);
}

ul {
  padding: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 2rem;
  list-style: none;
}

a {
  text-decoration: none;
  font-weight: 600;
  color: rgb(200, 200, 200);
}

.active {
  color: blueviolet;
}
