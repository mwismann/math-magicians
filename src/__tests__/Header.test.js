import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from '../components/Header';

test('Should render the Header', () => {
  const text = renderer.create(<BrowserRouter><Header /></BrowserRouter>);
  expect(text).toMatchSnapshot();
});
