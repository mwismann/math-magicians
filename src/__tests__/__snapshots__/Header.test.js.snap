// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Header 1`] = `
<header>
  <div
    className="logo-container"
  >
    <h1>
      Math Magicians
    </h1>
  </div>
  <nav>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
       | 
      <li>
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
       | 
      <li>
        <a
          className=""
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
